lib/options.nix

```diff
@@ -333,5 +333,120 @@ rec {
         else
           example;
     };
+
+  mkLazyLoadOption =
+    {
+      originalName,
+      lazyLoadDefaults ? { },
+    }:
+    lib.mkOption {
+      description = ''
+        Lazy-load settings for ${originalName}.
+      '';
+      type =
+        let
+          triggerType =
+            with types;
+            oneOf [
+              rawLua
+              str
+              (listOf str)
+            ];
+        in
+        types.submodule {
+          options = with defaultNullOpts; {
+
+            enable = lib.mkEnableOption ''
```

Comment:
While working with it, it just felt clunky to have to both enable AND select a backend otherwise no effect. I didn't want to automatically setup the config regardless of lazy loading attempt so we dont run into config breaking behavior if anything happens with the lazy modules and a user didn't enable the feature. But, I agree enable being a boolean is just simpler and then maybe we assert that you've enabled it but haven't selected a backend?

Reply:
I do like that

Reply:
We could also have a global `programs.nixvim.lazy-loading = { enabled = ....; };` option, and use the `config.lazy-loading.enable` as the default?

Reply:
That also works, or perhaps the default would be based on the global enable _and_ whether a lazyload trigger has been defined for this specific plugin?

Perhaps the exact defaults for the enable option are drifting out of scope for this PR then. As discussed https://github.com/nix-community/nixvim/pull/2602#discussion_r1866616042, the exact details of how to handle "globally" enabling lazyloading should be figured out after this PR.

Although we still need to think about those details a _little_ bit as they influence decisions like this one!

Reply:
I think that what we can say is that we will likely have a global toggle that will enable lazy-loading, and that plugins will by default opt-in into that, expect if they set _some set of TBD options_, in which case they will opt-out

Reply:
What I'm not sure about is do we want to support _different_ lazy providers for different plugins. I'm not sure about that, I think we should architecture the initial implementation as different providers exist, but that there will be a global option that allows to choose one

Reply:
I do have the per plugin backend option added, atm. I was about to remove it because it felt redundant if we just tell someone to enable a provider. But, if we want to support different providers per plugin, it becomes a good option again.

Reply:
I think we should only add it if someone comes with a compelling reason. I don't see any use for it

Reply:
Having multiple lazyload implementations at the same time seems pretty silly, but I guess it is a technical possibility.

Would it be unreasonable to have an assertion that only one is ever enabled? It'd certainly simplify things for us...

I personally think it makes the most sense to not have per-plugin "backend" options and instead use either "whatever backend plugin is enabled" or "whatever backend is configured, via a global backend option".

As for the per-plugin enable option: I'm wondering if it'd be intuitive for some plugins to default to `false` and others to default to `true` (depending on if we know a plugin is problematic when lazyloaded).

You would then need _both_ the global and per-plugin options enabled for it to be lazy-loaded.